File: Code Kata/maxsubarray.c
#include <stdio.h>
int main(void) 
{
	int n,sum=0,j=0;
	scanf("%d",&n);
    int arr[n],array[n],max=0;
    for(int i=0;i<n;i++)
    {
    	scanf("%d",&arr[i]);
    
    }
    for(int i=0;i<n;i++)
    {
    	if(arr[i]>max)
    	{
            	array[j]=i;
            	max=arr[i];
            	j++;
    	}
    }
    for(int i=array[0];i<=array[1];i++)
    {
    	sum+=arr[i];
    }
    
    printf("%d",sum);
	return 0;
}

File: Code Kata/maxsubarray.java
import java.io.*;
import java.util.*;
class Sidd
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner s=new Scanner(System.in);
		int n=s.nextInt();	
		int max=Integer.MIN_VALUE,min=0;
		int[] arr=new int[n];
		for(int i=0;i<n;i++)
		{
			arr[i]=s.nextInt();
		}
	for(int i=0;i<n;i++)
		{
		   min=min+arr[i];
		   if(max<min)
		    max=min;
		    if(min<0)
		    min=0;
		}
		System.out.println(max);
	}
}
